Modules/_asynciomodule.c

```diff
@@ -3723,7 +3723,9 @@ module_clear(PyObject *mod)
     Py_CLEAR(state->iscoroutine_typecache);
 
     Py_CLEAR(state->context_kwname);
-
+    // Clear the ref to running loop so that finalizers can run early.
+    // If there are other running loops in different threads,
+    // those get cleared in PyThreadState_Clear later.
```

Comment:
I think other threads actually get cleared earlier, but we probably don't need to be specific about the exact ordering in the comment. I think it's fine to say something like:

```
The running loops in other threads are cleared by PyThreadState_Clear.
```

I think the order of operations is:

1) Non-main threads may exit normally (and call PyThreadState_Clear()` as they exit)
2) Shutdown starts
3) Non-main threads (typically daemon threads) get cleared and deleted by `_PyThreadState_DeleteList()`
4) Modules get finalized and cleared
5) Finally, `PyThreadState_Clear()` is called on the main thread.